### DISCUSSION TITLE
How to deploy Astro SSR website?

### DISCUSSION THREAD

**Discussion by bautrukevich** [Dokploy/dokploy#152] [Q&A] [answered]
Hello!
I am using this recipe: https://docs.astro.build/en/recipes/docker/#multi-stage-build-using-ssr

And deploying using Dockerfile. But it seems that container with image is running and not forwarding ports like:
`docker run -p <local-port>:<container-port> <your-astro-image-name>`

And what I see is only:
```
Build dockerfile: ✅
Source Type: github: ✅
Step 1/17 : FROM node:lts AS base
 ---> d921256521a7
Step 2/17 : WORKDIR /app
 ---> Using cache
 ---> deeae206b033
Step 3/17 : COPY package.json package-lock.json ./
Docker Deployed: ✅
```

But what I am doing wrong? How to start application?

**Comment by Siumauricio** [comment]
When you deploying astro app need to set this astro config, need to expose the host

```ts
import { defineConfig } from "astro/config";
// https://astro.build/config
export default defineConfig({
  integrations: [react()],
  server: {
    port: 3000,
    host: true, // this line is important
  },
});
```

And then your domain should be something like this, make sure in the container port set 3000
![Screenshot 2024-06-20 at 8 05 51 PM](https://github.com/Dokploy/dokploy/assets/47042324/52ae1da6-73b5-4c51-a7f8-652be3254bdf)

**Comment by Siumauricio** [comment, marked as answer]
I create a example of astro website if you have still questions https://github.com/Dokploy/astro-example

**Reply by bautrukevich** [reply]
Thanks!
